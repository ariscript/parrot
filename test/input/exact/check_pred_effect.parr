def less_than_boxed(x, y, box):
    box[0] := box[0] + 1;
    x < y

check:
    example box = (0,),
    val less_than = (lambda (x, y): less_than_boxed(x, y, box)),
    1 is%(less_than) 2,
    box[0] is== 1,
    val _ = box[0] := 0,
    val shadow less_than = (lambda (x, y): let x = less_than(x, y) in print(box); x),
    1 is%(less_than) 3 because 2.

0
