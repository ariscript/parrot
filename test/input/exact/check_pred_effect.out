(1,)
(2,)
check block: 3/3 tests passed
0
